### PR TITLE
Harden events by tag stage spec

### PR DESCRIPTION
**Pull request by chbatey**
The test checks that the stage finds a first event in a previous bucket
but due to timing the test some times puts the event two buckets ago
due to the stage querying until the bucket after when the timeuuids
were generated.

This changes the test to hold off if it is within 2 seconds of a bucket
end.

Refs #278

### CODE REVIEW

**Comment by chbatey** [comment]
This also reduces the test time by ~10s with various config changes, hopefully they don't make it more flaky!

File: core/src/test/scala/akka/persistence/cassandra/query/EventsByTagStageSpec.scala [hunk 1]
```diff
@@ -51,8 +51,10 @@ object EventsByTagStageSpec {
        |   first-time-bucket = "${today.minusMinutes(5).format(firstBucketFormat)}"
        |   max-result-size-query = $fetchSize
        |   log-queries = on
-       |   refresh-interval = 250ms
-       |   events-by-tag-gap-timeout = 2s
+       |   refresh-interval = 200ms
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/cassandra/query/EventsByTagStageSpec.scala]: should we move the refresh-interval inside events-by-tag section or wait with that until we do the big config refactoring?

Comment (chbatey) [reply core/src/test/scala/akka/persistence/cassandra/query/EventsByTagStageSpec.scala]: It's used for events by persistence id as well so maybe not this time but we could add an override in this section later

Comment (patriknw) [reply core/src/test/scala/akka/persistence/cassandra/query/EventsByTagStageSpec.scala]: added a note for later https://github.com/akka/akka-persistence-cassandra/issues/81#issuecomment-348537776

**Pull request merged**